builder/dockerignore/dockerignore.go

```diff
@@ -33,3 +36,75 @@ func ReadAll(reader io.ReadCloser) ([]string, error) {
 	}
 	return excludes, nil
 }
+
+// ReadAllRecursive takes a base directory and a starting directory,
+// reads all .dockerignore files in subdirectories and returns the list of file
+// patterns to ignore. Note this will trim whitespace from each line as well
+// as use GO's "clean" func to get the shortest/cleanest path for each.
+func ReadAllRecursive(baseDir string, directory string) ([]string, error) {
```

Comment:
So the `ReadAllRecursive` method first reads the `.dockerignore` file at the current level, and then gets ready to recurse into subdirectories to do the deep scan. Prior to actually recursing, however, it does a check to make sure that the directory does not match any of the already-found exclusions. (This includes from previous levels). 

So, in the case that you mentioned, it would stop at the root, since every subsequent subdirectory is excluded.

I hope that answers your question?